cadl-extension/src/code-model-builder.ts

```diff
@@ -991,11 +993,12 @@ export class CodeModelBuilder {
 
     // properties
     for (const prop of type.properties.values()) {
-      if (prop.name === discriminatorPropertyName) {
-        // skip the discriminator property
+      if (
+        prop.name === discriminatorPropertyName || // skip the discriminator property
+        isNeverType(prop.type) // skip property of type "never"
```

Comment:
I thought "never" means it never gets to emitter?

Reply:
From what I learned, it means the variable is of not any type. Here I think it acts as a default fallback if no template argument is specified by user.

It's from `OperationStatus` of latest cadl-azure-core: https://github.com/Azure/cadl-azure/blob/97af7767bfd97280e3a25c48ed4f1ecda5d7f575/packages/cadl-azure-core/lib/foundations.cadl#L19-L33.

Reply:
But we seems to have provided the `TStatusResult` as `Resource`? Why it still have `never`?

```
  statusMonitor is CustomCore.PollingOperation<Resource>;

namespace CustomCore {
  @parentResource(T)
  @resource("operations")
  model OperationStatusResource<T> is Azure.Core.Foundations.OperationStatus<T>;

  @route("operations")
  op PollingOperation<TCustom> is Azure.Core.ResourceRead<OperationStatusResource<TCustom>>;
}```

Reply:
But you seem doing fine.

openapi3 https://github.com/microsoft/cadl/blob/main/packages/openapi3/src/openapi.ts#L1078-L1081

Reply:
Yes, it's odd... openapi3 seems not generating model for `Azure.Core.Foundations.OperationStatus<T>`.

But if I directly provide `TStatusResult`, it generates without `never`. I'll take another look.

https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAY2FkbC1sYW5nL3Jlc3QiOwrJGmF6dXJlLXRvb2xzL8UmxhFjb3JlIjsKCnVzaW5nIENhZGwuSHR0cDvMEVJlc3TIEUHENi5Db3JlOwoKQHNlcnZpY2VUaXRsZSgiTG9uZ1J1bm5pbmciKckdVmVyc2lvbigiMS4wLjAiKQpuYW1lc3BhY2XGXss2OwoKbW9kZWwgUmVzb3VyY2UgewogIEB2aXNpYmlsaXR5KCJyZWFkIikKICBpZDogc3RyxjYgIEBrZXnEB3NlZ21lbnTEKcZDc8UuxHTKMCAgdHlwyxB9yHhFeOQBLlBhcmFtc8Z8cXVlcsRZcHJvamVjdEZpbOgA0dk%2FZWTtAL%2FsAKkgIOgAl1Vyac1BQHJvdXRlKCIvbG9uZy1y6QFGaW50ZXJm5AEt6wEoT3DFWnN0YXR1c01vbml0b3IgaXMgQ3VzdG9t5AGdLlBvbGzFKmVyYXRpb2486ACUPuYBL3DPICjNXy7NW%2BQBQWNyZWF0ZU9yVXBkYXRlxG3LMchcQ80lz3NnZXTEN8gVUmVhZP8Alf8AlSAgZGVsZfkAjUTFHd9j32MgIGXlAdPsALlBY%2B0BUCztAjUs6gNDLkFjY2Vw5gIOcG9uc2UgJiBGb3VuZMV4c%2BwC%2BFPFdkxvY8Ub%2FwCq%2BACq5gPqePMAqwogICDJFCzFDvACqsYW%2FwC8%2FwC8xBsKICA%2B5QLQ6wPl6QKV5gNScGFyZW50yHkoVOQAvEDoAxcoIm%2FoAPDmA7nmA1npAQXGbsg9PFQ%2B5ADv6gSULuwApc81PFTnAVnnA2rPaW9wIPEDNFTmALzQZe0Cw%2FkAm8c85AEP